rules/cohort_hifiasm.smk

```diff
@@ -0,0 +1,137 @@
+        f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.hap2.p_ctg.lowQ.bed",
+        f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.hap2.p_ctg.noseq.gfa",
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.p_utg.gfa"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.p_utg.lowQ.bed"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.p_utg.noseq.gfa"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.r_utg.gfa"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.r_utg.lowQ.bed"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.r_utg.noseq.gfa"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.ec.bin"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.ovlp.reverse.bin"),
+        temp(f"cohorts/{cohort}/hifiasm/{{sample}}.asm.ovlp.source.bin")
+    log: f"cohorts/{cohort}/logs/hifiasm/{{sample}}.hifiasm.log"
+    benchmark: f"cohorts/{cohort}/benchmarks/hifiasm/{{sample}}.hifiasm.tsv"
+    conda: "envs/hifiasm.yaml"
+    params: prefix = f"cohorts/{cohort}/hifiasm/{{sample}}.asm"
+    threads: 48
+    message: "Executing {rule}: Assembling sample {wildcards.sample} from {input.fasta} and parental k-mers."
+    shell: "(hifiasm -o {params.prefix} -t {threads} -1 {input.parent1_yak} -2 {input.parent2_yak} {input.fasta}) > {log} 2>&1"
+
+
+rule gfa2fa:
+    input: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.gfa"
+    output: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.fasta"
+    log: f"cohorts/{cohort}/logs/gfa2fa/{{sample}}.asm.dip.{{infix}}.log"
+    benchmark: f"cohorts/{cohort}/benchmarks/gfa2fa/{{sample}}.asm.dip.{{infix}}.tsv"
+    conda: "envs/gfatools.yaml"
+    message: "Executing {rule}: Extracting fasta from assembly {input}."
+    shell: "(gfatools gfa2fa {input} > {output}) 2> {log}"
+
+
+rule bgzip_fasta:
+    input: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.fasta"
+    output: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.fasta.gz"
+    log: f"cohorts/{cohort}/logs/bgzip/{{sample}}.asm.dip.{{infix}}.log"
+    benchmark: f"cohorts/{cohort}/benchmarks/bgzip/{{sample}}.asm.dip.{{infix}}.tsv"
+    threads: 4
+    conda: "envs/htslib.yaml"
+    message: "Executing {rule}: Compressing {input}."
+    shell: "(bgzip --threads {threads} {input}) > {log} 2>&1"
+
+
+rule asm_stats:
+    input: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.fasta.gz"
+    output: f"cohorts/{cohort}/hifiasm/{{sample}}.asm.dip.{{infix}}.fasta.stats.txt"
+    log: f"cohorts/{cohort}/logs/asm_stats/{{sample}}.asm.dip.{{infix}}.fasta.log"
+    benchmark: f"cohorts/{cohort}/benchmarks/asm_stats/{{sample}}.asm.dip.{{infix}}.fasta.tsv"
+    conda: "envs/k8.yaml"
+    message: "Executing {rule}: Calculating stats for {input}."
+    shell: f"(k8 workflow/scripts/calN50/calN50.js -f {config['ref']['index']} {{input}} > {{output}}) > {{log}} 2>&1"
+
+
+rule align_hifiasm:
```

Comment:
We chatted about this offline, but might be nice in a future release to align each haplotype separately and use the parent name as part of the read group, then merge the results.